settings/ChangePassword/ChangePassword.js

```diff
@@ -0,0 +1,223 @@
+      fetch: false,
+      throwErrors: false,
+    },
+  });
+
+  static propTypes = {
+    stripes: PropTypes.shape({
+      intl: PropTypes.shape({
+        formatMessage: PropTypes.func.isRequired,
+      }).isRequired,
+      user: PropTypes.shape({
+        user: PropTypes.shape({
+          id: PropTypes.string,
+          firstName: PropTypes.string,
+          lastName: PropTypes.string,
+          username: PropTypes.string,
+        }).isRequired,
+      }).isRequired,
+    }),
+    mutator: PropTypes.shape({
+      changePassword: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    label: PropTypes.string.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      passwordMasked: true,
+    };
+
+    this.translateNamespace = 'ui-myprofile.settings.changePassword';
+    this.styles = {
+      changePasswordFormWrap: {
+        width: '100%',
+      },
+      toggleMaskButtonWrap: {
+        marginTop: '20px',
+      },
+    };
+  }
+
+  togglePasswordMask = () => {
+    this.setState(({ passwordMasked }) => ({
+      passwordMasked: !passwordMasked
+    }));
+  };
+
+  getFullUserName() {
+    const { user: { firstName, lastName } } = this.props.stripes.user;
+
+    return { firstName, lastName };
+  }
+
+  onChangePasswordFormSubmit = (values, dispatch, { reset }) => {
+    const { changePassword } = this.props.mutator;
+    const { currentPassword, newPassword } = values;
+    const { username, id: userId } = this.props.stripes.user.user;
+
+    return changePassword
+      .POST({
+        username,
+        userId,
+        password: currentPassword,
+        newPassword,
+      })
+      .then(() => {
+        this.handleChangePasswordSuccess();
+        reset();
+      })
+      .catch(this.handleChangePasswordError);
+  };
+
+  handleChangePasswordSuccess = () => {
+    const fullUserName = this.getFullUserName();
+
+    const successMessage = (
+      <SafeHTMLMessage
+        id={`${this.translateNamespace}.successfullyChanged`}
+        values={fullUserName}
+      />
+    );
+
+    this.callout.sendCallout({ message: successMessage });
+  };
+
+  handleChangePasswordError = err => {
+    const isWrongCurrentPassword = err.status === 401;
+
+    if (isWrongCurrentPassword) {
+      throw new SubmissionError({
+        currentPassword: this.translate('wrongPassword')
+      });
+    }
+  };
+
+  validateForm = values => {
+    const errors = {};
+    const enterValueError = this.translate('enterValue');
+
+    if (!values.currentPassword) {
+      errors.currentPassword = enterValueError;
+    }
+
+    if (!values.newPassword) {
+      errors.newPassword = enterValueError;
+    }
+
+    if (!values.confirmPassword) {
+      errors.confirmPassword = enterValueError;
+    }
+
+    const isFormValid = (
+      values.newPassword &&
+      values.confirmPassword &&
+      values.newPassword !== values.confirmPassword
```

Comment:
Should this be testing for equality rather than inequality?

Reply:
Or is the condition right, but the variable name `isFormValid` misleading?

Reply:
Nice catch - will rename it.